restorer_test.go

```diff
@@ -19,23 +20,33 @@ import (
 	"github.com/buildpacks/lifecycle/cache"
 	"github.com/buildpacks/lifecycle/cmd"
 	"github.com/buildpacks/lifecycle/layers"
+	"github.com/buildpacks/lifecycle/platform"
 	h "github.com/buildpacks/lifecycle/testhelpers"
 )
 
 func TestRestorer(t *testing.T) {
 	for _, buildpackAPI := range api.Buildpack.Supported {
-		spec.Run(t, "unit-restorer/buildpack-"+buildpackAPI.String(), testRestorerBuilder(buildpackAPI.String()), spec.Report(report.Terminal{}))
+		buildpackAPIStr := buildpackAPI.String()
+		for _, platformAPI := range api.Platform.Supported {
```

Comment:
I was thinking about this combinatorial explosion, and I want to say that this might be resolved if we do https://github.com/buildpacks/lifecycle/issues/587 AND we pull the buildpack-api-related logic in restorer ([finding the cache layers](https://github.com/buildpacks/lifecycle/blob/3dc792afcfb613b0ebd5e32bb5204387e319c2a4/restorer.go#L44-L57)) into a separate service that itself could be unit-tested. Then we could keep the platform-related logic at the top level, like we do for [build](https://github.com/buildpacks/lifecycle/blob/3dc792afcfb613b0ebd5e32bb5204387e319c2a4/builder.go#L73) and [detect](https://github.com/buildpacks/lifecycle/blob/3dc792afcfb613b0ebd5e32bb5204387e319c2a4/detector.go#L119). WDYT?

Reply:
Given that we're eager to merge this PR I'd be okay deferring this to #587 or some other issue, just raising it for consideration.